src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -19,9 +19,17 @@ public class AuthenticationSchemeProvider : IAuthenticationSchemeProvider
         /// </summary>
         /// <param name="options">The <see cref="AuthenticationOptions"/> options.</param>
         public AuthenticationSchemeProvider(IOptions<AuthenticationOptions> options)
+            : this(options, StringComparer.Ordinal)
```

Comment:
Instead of passing the comparer, can we just have the ctor take the map instance instead?

i.e. `protected AuthenticationSchemeProvider(IDictionary<string, AuthenticationScheme> map)`  this seems generally more flexible than a specific comparer behavior.

Reply:
I made the change, but I'm not so sure about it... It feels kinda weird passing in a dictionary, just for the purpose of changing the `StringComparer`. I.e. am I supposed to add something to it? Am I *allowed* to add something to it? What happens if I do? What happens if I keep a reference to it and mutate it?

Reply:
Yep its pretty powerful, in this case, you hold the source of truth, so if you mutate that reference, you are directly accessing the underlying dictionary backing the provider.  Seems fine to give developers this power though since its a protected ctor, this is something they are responsible for using 'wisely' :)

Maybe instead of calling it map, name the parameter something like schemeCache to make it clear that this is what will be used to serve the data

Reply:
Another thing is that the dictionary is accessed within locks inside the `AuthenticationSchemeProvider` (at least in some places). Is that something we should care about?

Reply:
Its probably worth mentioning in the doc comments.  Basically the scheme provider locks to prevent concurrency issues when modifying the list of schemes, so there's definitely issues if someone is back door modifying/accessing the underlying dictionary directly.

Reply:
But basically its the `IAuthenticationSchemeProviders` job to manage the schemes, the dictionary passed in is store backing the provider, so you can probably just say its best to always go through the interface instead of modifying the dictionary.